controllers/ccruntime_controller.go

```diff
@@ -629,15 +629,16 @@ func (r *CcRuntimeReconciler) processDaemonset(operation DaemonOperation) *appsv
 	}
 
 	var runtimeClassNames []string
-	var snapshotter = ""
+	var snapshotter_handler_mapping = ""
 	for _, runtimeClass := range r.ccRuntime.Spec.Config.RuntimeClasses {
 		runtimeClassNames = append(runtimeClassNames, runtimeClass.Name)
-		// FIXME: This will need to be changed by the moment the kata-containers
-		//        payload script supports setting one snapshotter per runtime handler.
-		//        For now, for the v0.8.0 release, we're fine assuming that all the
-		//        set snapshotters are going to be the same.
-		if snapshotter == "" && runtimeClass.Snapshotter != "" {
-			snapshotter = runtimeClass.Snapshotter
+
+		if runtimeClass.Snapshotter != "" {
```

Comment:
I think that here it should be `== ""`.

One alternative is to use Go map, convert to json then to string:
```
snapshotters := map[string]string{"kata-qemu": "nydus", "kata-clh": "nydus"}
j, _ := json.Marshal(snapshotters)
snapshottersStr := string(j)
```

But then `SNAPSHOTTER_HANDLER_MAPPING` should be processed as a json string on kata side (i.e. `{"k1":"v1","k2":"v2"}` vs  `"k1":"v1","k2":"v2"`). One advantage of passing a json string is that in the future we can pass properties alongside if needed.

Reply:
Wainer is correct - the current logic produces:
`Validating the snapshotter-handler mapping: ",kata-clh:nyduskata-qemu:nyduskata-qemu-tdx:nyduskata-qemu-sev:nyduskata-qemu-snp:nydus"`. I'll try and fix it, rebase and push to this branch

Reply:
I managed to force push in the end to hopefully fix this.